button-listen.c

```diff
@@ -11,37 +14,77 @@ static double timespec_diff(struct timespec const* start, struct timespec const*
     return s + (ns / 1e9);
 }
 
+static bool open_file(FILE **evf, const char *filename)
+{
+    FILE *file = NULL;
+    int fd = -1;
+    file = fopen(filename, "rb");
```

Comment:
`"re"`, binary mode is meaningless on POSIX, and we usually prefer CLOEXEC by default (and there's no compatibility issue with that here).

Reply:
Ah sorry. I had preserved that from the original.